loopd/daemon.go

```diff
@@ -161,13 +152,28 @@ func daemon(config *config, lisCfg *listenerCfg) error {
 		log.Infof("Waiting for updates")
 		for {
 			select {
-			case swap := <-statusChan:
+			case swapInfo := <-statusChan:
 				swapsLock.Lock()
-				swaps[swap.SwapHash] = swap
+
+				updateFn := swapClient.Store.UpdateLoopIn
+				if swapInfo.SwapType == swap.TypeOut {
+					updateFn = swapClient.Store.UpdateLoopOut
+				}
+				log.Debugf("Updating swap %s on disk",
+					swapInfo.SwapHash)
+				err := updateFn(
```

Comment:
hm, where/when did we persist before this change? 🤔

Reply:
I thought that some intermediate states were not persisted as the daemon would go through the swap on startup and find out what the state was. But as it turns out, persisting all updates is a bad idea when I found this comment:
> If an unexpected error happened, report a temporary failure
> but don't persist the error. Otherwise for example a
> connection error could lead to abandoning the swap
> permanently and losing funds.

So I removed the whole persist logic here again. This probably means that the REST interface won't get the same granularity of swap states as the monitor stream. But I think this is less of a problem compared to unwanted side effects.

@joostjager could you please take a look at this anyway?

Reply:
Yes, that is indeed the idea. We don't store these temp failures in the database. It could just be a lost connection for example. But I do think it is useful to expose these to rpc users, because how do you otherwise find out that the swap (temporarily) failed?

Reply:
I see your point. But having this global state also doesn't feel too nice.
Maybe we should try to make REST users 1st class citizens by allowing them to use streaming RPCs as well with something like https://github.com/tmc/grpc-websocket-proxy .

Reply:
Does this have anything to do with streaming vs polling? Also for polling, they'd probably want to receive a temp. failure if it happened

Reply:
I think this regression is something to be sure we want to say yes to.

Reply:
Yes, you are right. I was too hung up on the "no global state" that I kind of missed the middle ground that solves both problems (see last comment).